source/steamshim/src/os.cpp

```diff
@@ -20,8 +20,19 @@ freely, subject to the following restrictions:
 
 #include "os.h"
 #include "parent/parent_private.h"
+#include <cstdarg>
+#include <stdarg.h>
+#include <stdio.h>
 #include <cstring>
 
+bool debug = false;
+void dbgprintf(const char *fmt, ...){
+    if (!debug) return;
+    va_list va;
+    va_start(va, fmt);
+    vprintf(fmt, va);
```

Comment:
https://linux.die.net/man/3/vprintf
```
The functions vprintf(), vfprintf(), vsprintf(), vsnprintf() are equivalent to the functions printf(), fprintf(), sprintf(), snprintf(), respectively, except that they are called with a va_list instead of a variable number of arguments. These functions do not call the va_end macro. Because they invoke the va_arg macro, the value of ap is undefined after the call. See [stdarg](https://linux.die.net/man/3/stdarg)(3).
```

you need to call va_end?